pages/search/index.js

```diff
@@ -1,48 +1,285 @@
+          <SearchInput
+            type="text"
+            id="text"
+            placeholder="Search"
+            value={searchQuery}
+            onChange={(event) => setSearchQuery(event.target.value)}
+          />
+
+          <StyledButton
+            type="button"
+            onClick={() => {
+              setSearchQuery("");
+              
+              setShowFilters(true);
+            }}
+          >
+            Advanced Search
+          </StyledButton>
+        </>
+      )}
+
+      {showFilters && (
+        <>
+          <StyledForm onSubmit={handleSubmit}>
+            <CloseButton
+              onClick={() => {
+                setShowFilters(false);
+                setFilters(INITIAL_FILTERS);
+              }}
+            />
+            <h2>Advanced Search</h2>
+            <Label htmlFor="name">Name</Label>
+            <FormInput
+              type="text"
+              name="name"
+              id="name"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="Exact name"
+            />
+            <Label htmlFor="categories">Category</Label>
+            <FormInput
+              type="text"
+              name="categories"
+              id="categories"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="Genre, theme, etc."
+            />
+            <Label htmlFor="rating">Rating</Label>
+            <FormInput
+              type="number"
+              name="rating"
+              id="rating"
+              step="0.1"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="(0-10)"
+            />
+            <Label htmlFor="minAge">Minimum Age</Label>
+            <FormInput
+              type="number"
+              name="minAge"
+              id="minAge"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="(1-99)"
+            />
+            <Label htmlFor="players">Players</Label>
+            <FormInput
+              type="number"
+              name="players"
+              id="players"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="min. 1"
+            />
+            <Label htmlFor="playtime">Playtime</Label>
+            <FormInput
+              type="number"
+              name="playtime"
+              id="playtime"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="in minutes"
+            />
+            <Label htmlFor="yearpublished">Release Year</Label>
+            <FormInput
+              type="number"
+              name="yearpublished"
+              id="yearpublished"
+              defaultValue={""}
+              onChange={handleInputChange}
+              placeholder="yyyy"
+            />
+
+            <StyledButton type="submit">Submit</StyledButton>
```

Comment:
It's `type="submit` by default. 
What's the sense of it? Since you search while typing you don't have to submit anything.

Reply:
The goal was to show the results only after clicking Submit. We didn't remove it to show the intention but we couldn't figure a way to do this without using useState